src/pages/items/hooks/usePageSize.jsx

```diff
@@ -0,0 +1,23 @@
+import useViewport from 'shared/hooks/useViewport';
+
+function usePageSize(type) {
+    const { deviceType } = useViewport();
+
+    const getPageSize = (type) => {
+        if (type === 'all') {
+            if (deviceType === 'mobile') return 4;
+            if (deviceType === 'tablet') return 6;
+            if (deviceType === 'desktop') return 10;
+        }
+        if (type === 'best') {
+            if (deviceType === 'mobile') return 1;
+            if (deviceType === 'tablet') return 2;
+            if (deviceType === 'desktop') return 4;
+        }
+    };
```

Comment:
코드의 길이는 조금 더 길어지지만, 아래와 같이 반복 부분을 제거하고 기본값을 제공하는 것이 안정성 측면에서 장점이 있습니다.

```suggestion
const getPageSize = (type) => {
    switch (type) {
        case 'all':
            switch (deviceType) {
                case 'mobile':
                    return 4;
                case 'tablet':
                    return 6;
                case 'desktop':
                    return 10;
                default:
                    return 10; // 기본값 설정
            }
        case 'best':
            switch (deviceType) {
                case 'mobile':
                    return 1;
                case 'tablet':
                    return 2;
                case 'desktop':
                    return 4;
                default:
                    return 4; // 기본값 설정
            }
        default:
            return 10; // 기본값 설정
    }
};

```